### ISSUE TITLE
Can States run independently of FSMM?

### ISSUE THREAD

**Issue by MyWorldNetwork** [Fexcraft/States#13]
I really like and wish to use the States mod however the FSMM mod is just confusing and illogical. In addition there seems to be no real reason to force a dependency on an economy mod, this should instead be an optional dependency.

I cannot accurately create an economy on my server with FSMM in its current state but lack the skills to recode it myself.

If FSMM cannot be made to work correctly, ie to actually adhere to proper and accurate decimalisation I would instead prefer to use a different 'bank' mod, which is a genuine shame as i genuinely like and would prefer  to use FSMM along with States.

**Comment by ProsperCraft** [comment]
Lol we have been round and round about that decimal, it doesn't make sense at all, agreed.

**Comment by MyWorldNetwork** [comment]
Even if someone can explain it to me so that I can understand how it works but at the moment the command /money tells me I have what looks like $100 and the ATM tells me I have $100,000...

Getting rid of 'cents' would be a huge improvement and go some way to it making some semblance of logical sense.

**Comment by Fexcraft** [comment]
Code wise it's actually centesimals, and are there even "different" bank mods?

To the original question, how would States work payment wise without FSMM?
Everything would have to cost nothing, and tax would be non existent.

**Comment by Fexcraft** [comment]
And are there even "different" bank mods?

**Comment by MyWorldNetwork** [comment]
I have no idea but if there aren't and we cannot get FSMM to work correctly then i cannot use it (States).

Can you explain what centesimal is and how this is broken down into decimals so that I can work out how to price things?

If we can resolve this i'm genuinely more than happy to thoroughly test the mod out as it seems ideal otherwise.

**Comment by Fexcraft** [comment]
What's the issue with FSMM?
Generally 1000 input in either commands or atm translates to 1 F$.
1000 = 1
1200290 = 120,29

Only thing that I'd guess is confusing is that `,` and `.` in some parts of the world have the opposite meaning, but there is a config in FSMM to switch it around.
Then there is another config if it should show only decimals or the centesimals aswel.

Those settings for part do not sync (comma/dot) or apply(decimal display) to the ATM GUI,
as far I recall the ATM will always show the centesimals.
I might see about fixing (also adding config sync upon joining a server) that in the future,
but it won't be changed that 1000 equals one unit.

It's for allowing currecies which go up to centesimals, or more fine tuned calculations for taxes or other bank activities, once States get's the `"Companies"` features.

**Comment by MyWorldNetwork** [comment]
I think the main problem is that it's more complex than it needs to be.

If you understand centesimals, and I have to be honest, even with coming up on 20 years in financial services, i have never heard the term, then i guess the whole thing must seem very trivial but your clarification that 1200290 = 120.29 doesn't make sense to me!

**Comment by Fexcraft** [comment]
uff, so let's explain it so.
There is the `float` datatype, it supports numbers on the decimal point,
but often on calculations it ends up "inaccurate", it outputs things like 0.000001 cent.
Then there is `double` datatype, supposedly it's double as accurate, in older FSMM versions I used that.
Still, there been situations when I still got such "errors".

I ended up deciding to use `long` datatype. It has not such issues at all,
the downside is the lack of decimal point numbers.
So I decided to go via the rule of 1000 = 1, a third 0 for the sake of I forgot why.

As such the commands and atm expect the "real" number, not the formatted balance.
I always thought putting e.g. "105,2" into the ATM, and let it translate it to "105200" is too risky.
I might revise that though.

Lemme write those things down on trello.
_edit_
https://trello.com/c/S66MoDg3
https://trello.com/c/LgfdDPEK

**Comment by MyWorldNetwork** [comment]
Thanks.

My server is not publicly advertised and therefore has a playerbase of 1-2 so no great impact. I'll continue trying to puzzle out this whole 'centemisals' in an effort to be able to explain in layman's terms to any future players.

**Comment by Fexcraft** [comment]
So, config sync `Server -> Client` upon joining should be working (not tested).

And added an optional text field for keyboard input.
https://cdn.discordapp.com/attachments/365126848914391040/652276429488390163/2019-12-05_23-29-08.mp4
This should solve usual issues with the number buttons 🤔 
Also what's not visible on the video, right after I also removed the forced-on centesimals on the result/response screen in the UI.

The centesimals at the normal input remained, but it's easy to bypass by simply enabling the text field.

~~_I didn't read the whole old conversation, I did based on Trello notes._~~
